entities/PropertyManagementService.js

```diff
@@ -0,0 +1,199 @@
+        .filter((p) => p.group === boardProperty.group)
+        .filter((p) => p.ownedBy === playerIndex).length;
+      rentAmount = railroadPricing[railroadCount - 1];
+    } else {
+      if (boardProperty.buildings > 0) {
+        rentAmount = boardProperty.multipliedRent[boardProperty.buildings - 1];
+      } else {
+        const hasMonopoly = this.hasMonopoly(
+          gameState,
+          boardProperty.group,
+          playerIndex
+        );
+
+        if (hasMonopoly) {
+          rentAmount *= 2;
+        }
+      }
+    }
+    return rentAmount;
+  }
+
+  static changeOwner(boardProperty, playerId) {
+    boardProperty.ownedBy = playerId;
+  }
+
+  static renovate(gameState, property) {
+    gameState.config.propertyConfig[
+      property.buildings !== 4 ? 'houses' : 'hotels'
+    ] -= 1;
+    property.buildings += 1;
+    WealthService.buyAsset(gameState.currentPlayer, property.houseCost);
+  }
+
+  static demolish(gameState, property) {
+    gameState.config.propertyConfig[
+      property.buildings !== 5 ? 'houses' : 'hotels'
+    ] += 1;
+    property.buildings -= 1;
+    WealthService.sellAsset(gameState.currentPlayer, property.houseCost / 2);
+  }
+
+  static toggleMortgageOnProperty(
+    gameState,
+    boardProperty,
+    player = gameState.currentPlayer
+  ) {
+    boardProperty.mortgaged = !boardProperty.mortgaged;
+    const INTEREST_RATE = 0.1;
+    const mortgageBaseCost = boardProperty.price / 2;
+
+    if (boardProperty.mortgaged) {
+      WealthService.sellAsset(player, mortgageBaseCost);
+    } else {
+      WealthService.decrement(player, mortgageBaseCost * INTEREST_RATE);
+      WealthService.buyAsset(player, mortgageBaseCost);
+    }
+  }
+
+  static hasMonopoly(gameState, propertyGroup, playerId) {
+    return this.getProperties(gameState)
+      .filter((p) => p.group === propertyGroup)
+      .every((p) => p.ownedBy === playerId);
+  }
+
+  static getAvailableManagementActions(gameState) {
+    let availableActions = ['TOGGLE_MORTGAGE'];
+
+    if (this.getRenoProperties(gameState).length > 0) {
+      availableActions.push('RENOVATE');
+    }
+
+    if (this.getDemoProperties(gameState).length > 0) {
+      availableActions.push('DEMOLISH');
+    }
+
+    availableActions.push('CANCEL');
+    return availableActions;
+  }
+
+  static getRenoProperties(gameState) {
+    const { houses, hotels, properties } = gameState.config.propertyConfig;
+    const player = gameState.currentPlayer;
+    if (houses + hotels === 0) return [];
+    const limitedHouses = houses === 0;
+    const limitedHotels = hotels === 0;
+
+    const availableRenoProps = _(properties)
+      .filter((p) => p.ownedBy === player.id)
+      .groupBy((p) => p.group) // { 'groupName': [properties] }
+      .mapValues((properties) => ({
+        properties,
+        hasMonopoly: this.hasMonopoly(
+          gameState,
+          properties[0].group,
+          properties[0].ownedBy
+        ),
+        minBuildingCount: Math.min.apply(
```

Comment:
Using spread should simplify this to `Math.min(...properties.map(p => p.buildings))` ?

Reply:
ooo nice optimization!